addOns/openapi/src/main/java/org/zaproxy/zap/extension/openapi/OpenApiAPI.java

```diff
@@ -138,6 +127,17 @@ public ApiResponse handleApiAction(String name, JSONObject params) throws ApiExc
         }
     }
 
+    private static ApiResponseList createResponse(String name, List<String> errors) {
+        ApiResponseList result = new ApiResponseList(name);
+        for (String error : errors) {
+            result.addItem(new ApiResponseElement("warning", error));
+        }
```

Comment:
The issue is suggesting throwing an API exception when there are errors, to return the 400 code (and `ApiResponseElement.OK` when ok).

Reply:
I've adjusted this to be closer to what's needed. However, am I overlooking a way to return a list but still set 400? (Should I update core to allow setting the status code?)